controller_manager:
  ros__parameters:
    update_rate: 100  # Hz
    use_sim_time: true

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

joint_state_broadcaster:
  ros__parameters:
    publish_rate: 50

joint_trajectory_controller:
  ros__parameters:
    joints:
      - joint_0
      - joint_1
      - joint_2
    command_interfaces:
      - position
    state_interfaces:
      - position
    allow_partial_joints_goal: false # Defaults to false
    open_loop_control: false
    allow_integration_in_goal_trajectories: false
    constraints:
      stopped_velocity_tolerance: 0.001 # Defaults to 0.01
      goal_time: 0.0 # Defaults to 0.0 (start immediately)   
    gains:
        joint_0  : {p: 9, i: 0.03, d: 0.3055, i_clamp_min:  -1000, i_clamp_max: 1000}
        joint_1  : {p: 9, i: 0.03, d: 0.3055, i_clamp_min:  -1000, i_clamp_max: 1000}
        joint_2  : {p: 9, i: 0.03, d: 0.3055, i_clamp_min:  -1000, i_clamp_max: 1000}